### PR TITLE
[WIP] Extended Weekly Schedule Widget

### PR DESCRIPTION
**Pull request by xabispacebiker**
This pull request improves the weekly widget as in https://97irratia.info/programazioa/

https://discourse.libretime.org/t/fast-new-extended-weekly-schedule-widget-for-testing/133/4

### CODE REVIEW

File: airtime_mvc/public/widgets/widget_schedule_v3.html [hunk 1]
```diff
@@ -0,0 +1,21 @@
+<link rel="stylesheet" href="/airtime-widgets/css/airtime-widgets.css" type="text/css" />
+<link rel="stylesheet" href="/airtime-widgets/css/custom.css" type="text/css" />
+<script src="/airtime-widgets/js/jquery-1.6.1.min.js" type="text/javascript"></script>
+<script src="/airtime-widgets/js/jquery-ui-1.8.10.custom.min.js" type="text/javascript"></script>
+<script src="/airtime-widgets/js/jquery.showinfo.js" type="text/javascript"></script>
+<script>
+    $(document).ready(function() {
+        $("#scheduleTabs").airtimeWeekSchedule({
+            sourceDomain:"https://airtime.97irratia.info/",
```
Comment (hairmare) [inline airtime_mvc/public/widgets/widget_schedule_v3.html]: I think we'll need to make this URI dynamic.

File: airtime_mvc/public/widgets/widget_schedule_v3.html [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+            dowText:{monday:"Monday", tuesday:"Tuesday", wednesday:"Wednesday", thursday:"Thursday", friday:"Friday", saturday:"Saturday", sunday:"Sunday", nextmonday:">> Monday", nexttuesday:">> Tuesday", nextwednesday:">> Wednesday", nextthursday:">> Thursday", nextfriday:">> Friday", nextsaturday:">> Saturday"},
+            miscText:{time:"Time", programName:"Show name", details:"Details", readMore:"Read more", defaultGenre:"Radio", nowPlaying:"on Air now..."},
+	    backgroundColor: "#d52a33",
+	    textColor: "#fff",
+	    programImage: "https://97irratia.info/wp-content/uploads/2015/03/6461315162291g.jpg",
```
Comment (hairmare) [inline airtime_mvc/public/widgets/widget_schedule_v3.html]: This one as well...

File: airtime_mvc/public/widgets/widget_schedule_v3.html (hunk 1 above)
Comment (xabispacebiker) [reply airtime_mvc/public/widgets/widget_schedule_v3.html]: Ups, yeah.. nice catch

**Comment by Robbt** [comment]
I suspect that the issue might be that the commit you are trying to undo is already committed to your master branch or whatever you are forking from. I know I've seen unrelated commits show up when I didn't have my master synced to the main libretime master repo. I don't know I'm not a git master, but hopefully we can figure this out.

I was also thinking that ideally we would have some sort of choice for people, like different widgets/themes they could use, but that is outside of the scope of this commit.

**Comment by hairmare** [comment]
Hey @xabispacebiker I think I realized what went wrong. Do you have a local working copy? If so, could you please post the output of `git remote -v`? I'll write up some step by step instructions to get you recovered, but you'll need to have a local working copy to follow them.

**Comment by xabispacebiker** [comment]
> Hey @xabispacebiker I think I realized what went wrong. Do you have a local working copy? If so, could you please post the output of `git remote -v`? I'll write up some step by step instructions to get you recovered, but you'll need to have a local working copy to follow them.

Hi @hairmare , I do.. this is the output:

origin	https://xabierarrabal%40airean.info@github.com/xabispacebiker/libretime (fetch)
origin	https://xabierarrabal%40airean.info@github.com/xabispacebiker/libretime (push)

**Comment by xabispacebiker** [comment]
I am thinking of removing this pull request as I have realized there is another one which is more integrated with libretime https://github.com/LibreTime/libretime/pull/443

**Comment by Robbt** [comment]
I wouldn't remove it. We may want to create a PR that provides multiple options as a UI choice or takes the best of both of these PRs and combines them.

**Comment by hairmare** [comment]
@xabispacebiker The po change seems to be on your master branch. I would probably recover from this as follows, just be sure to copy the airtime.po file somewhere safe as the below will remove you the change from git.

```bash
# switch to master and revert to upstream
git checkout master
git remote add upstream  git@github.com:LibreTime/libretime.git
git fetch --all
git reset upstream/master --hard
git push origin --force

# rebase libretime-widgets-v3 branch to updated master
git checkout libretime-widgets-v3
# rename working branch
git branch -m libretime-widgets-v3-old
# recreate branch with libretime-widgets-v3 name
git checkout -b libretime-widgets-v3
# cherry-pick all the commit except the po update onto new branch
git cherry-pick 210e7d34bf765656bad237298114d9c2fb4f1f9e d23a4d8a6256949b868c951de5fd007dfc7dce7d a5bced468324a7ef763951272e2cf86b8c914d53 6151068a53ae1ecb67f43c7378f4befc89a06d8a b3e771534466bdebec27fc33478c5b7ba68e0aa4
# force push new branch into this PR
git push origin libretime-widgets-v3 --force-with-lease
```

After this you can create a new commit with the airtime.po change by switching to a new branch after checking out master and redoing the commit.

In the future, when you want to update your master branch you should do it like so:
```
git checkout master
git pull upstream master
```

Usually we recommend not putting any changes onto the master branch and keeping it in sync with upstream like above.

**Comment by frecuencialibre** [comment]
i hit this same problem on a much simpler pull request. i had to stash changes, delete the branch both locally and in my fork, and then create a new branch for my PR, this time _from the last commit pulled from origin_ :
```
git branch my_fabulous_contribution <commit_hash>
```

**Comment by JohnnyC1951** [comment]
I made my version  I now run it from a folder the public folder. The makes for easy site styling without mangling the core.

**Comment by xabispacebiker** [comment]
Definitely this pullrequest is not the best way to do it and needs some improvements, somebody requested in the forum to upload it to libretime but I know it can be done better. 
@JohnnyC1951 , I used to do what you do in a cron job, I didn't have the website server on the same server running libretime, so I had to include an ftp command to upload the generated schedule in a frequently running cron, probably similar as yours. 

I think It would be a nice implementation disallowing the direct access to web widgets and process all requests via cached files so the server does not get overloaded in case of many schedule or "on air" widget requests, the server gets exhausted when this happens.

**Comment by JohnnyC1951** [comment]
I just pull the schedule widget, using a one-line bash cron script that calls wget and saves it to file several times an hour. This could be on the station's listener website. It could all be done in the widget without a cron (if file is older than 20 mins then curl a new copy into the cache), The listener facing website simply loads that cached file into an iframe. Not ideal, but it works and is simple, very fast and easy. Maybe we need a generic celery thing to take the place of the cron, meanwhile cron works great.

**Pull request closed without merging**